pkg/extensions/search/cve/cve.go

```diff
@@ -1,205 +1,202 @@
+				break
+			}
+		}
 
-	var rootDir string
+		if hasCVE {
+			infectedTags = append(infectedTags, tagInfo)
+		}
+	}
 
-	// Get corresponding CVE trivy config, if no sub cve config present that means its default
-	trivyCtx, ok = cveinfo.CveTrivyController.SubCveConfig[prefixName]
-	if ok {
-		imgStore := cveinfo.StoreController.SubStore[prefixName]
+	if len(infectedTags) != 0 {
+		cveinfo.Log.Info().Str("repo", repo).Msg("comparing fixed tags timestamp")
 
-		rootDir = imgStore.RootDir()
+		tagsInfo = common.GetFixedTags(tagsInfo, infectedTags)
 	} else {
-		trivyCtx = cveinfo.CveTrivyController.DefaultCveConfig
-
-		imgStore := cveinfo.StoreController.DefaultStore
-
-		rootDir = imgStore.RootDir()
+		cveinfo.Log.Info().Str("repo", repo).Str("cve-id", cveID).
+			Msg("image does not contain any tag that have given cve")
 	}
 
-	trivyCtx.Input = path.Join(rootDir, image)
-
-	return trivyCtx
+	return tagsInfo, nil
 }
 
-func (cveinfo CveInfo) GetImageListForCVE(repo, cvid string, imgStore storage.ImageStore,
-	trivyCtx *TrivyCtx,
-) ([]ImageInfoByCVE, error) {
-	imgList := make([]ImageInfoByCVE, 0)
+func (cveinfo BaseCveInfo) GetCVEListForImage(image string) (map[string]cvemodel.CVE, error) {
+	cveMap := make(map[string]cvemodel.CVE)
 
-	rootDir := imgStore.RootDir()
-
-	manifests, err := cveinfo.LayoutUtils.GetImageManifests(repo)
-	if err != nil {
-		cveinfo.Log.Error().Err(err).Msg("unable to get list of image tag")
-
-		return imgList, err
+	isValidImage, err := cveinfo.Scanner.IsImageFormatScannable(image)
+	if !isValidImage {
+		return cveMap, err
 	}
 
-	for _, manifest := range manifests {
-		tag := manifest.Annotations[ispec.AnnotationRefName]
-
-		image := fmt.Sprintf("%s:%s", repo, tag)
-
-		trivyCtx.Input = path.Join(rootDir, image)
-
-		isValidImage, _ := cveinfo.LayoutUtils.IsValidImageFormat(image)
-		if !isValidImage {
-			cveinfo.Log.Debug().Str("image", repo+":"+tag).Msg("image media type not supported for scanning")
-
-			continue
-		}
-
-		cveinfo.Log.Info().Str("image", repo+":"+tag).Msg("scanning image")
-
-		report, err := ScanImage(trivyCtx.Ctx)
-		if err != nil {
-			cveinfo.Log.Error().Err(err).Str("image", repo+":"+tag).Msg("unable to scan image")
-
-			continue
-		}
-
-		for _, result := range report.Results {
-			for _, vulnerability := range result.Vulnerabilities {
-				if vulnerability.VulnerabilityID == cvid {
-					digest := manifest.Digest
+	return cveinfo.Scanner.ScanImage(image)
+}
 
-					imageBlobManifest, err := cveinfo.LayoutUtils.GetImageBlobManifest(repo, digest)
-					if err != nil {
-						cveinfo.Log.Error().Err(err).Msg("unable to read image blob manifest")
+func (cveinfo BaseCveInfo) GetCVESummaryForImage(image string) (ImageCVESummary, error) {
+	imageCVESummary := ImageCVESummary{
+		Count:       0,
+		MaxSeverity: "UNKNOWN",
+	}
 
-						return []ImageInfoByCVE{}, err
-					}
+	isValidImage, err := cveinfo.Scanner.IsImageFormatScannable(image)
+	if !isValidImage {
+		return imageCVESummary, err
```

Comment:
We should cover this. I think there should be a test case already for that.

Reply:
Will do. This is actually the new method I added to count the CVEs. Will look at the test for `GetCVEListForImage` which is the older method and has coverage.